### PR TITLE
feat: viem

### PR DESCRIPTION
**Pull request by TateB**

### CODE REVIEW

File: docs/public/function.getAvailable.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+> **getAvailable**(`client`, `parameters`): `Promise`\< `boolean` \>
+
+Gets the availability of a name to register
+
+## Example
+
+```ts
+import { createPublicClient, http } from 'viem'
+import { mainnet } from 'viem/chains'
+import { addEnsContracts, getAvailable } from '@ensdomains/ensjs'
```
Comment (LeonmanRolls) [inline docs/public/function.getAvailable.md:18]: In nextjs started I couldn't actually import these public functions form ensjs, had to use public/ensjs

File: docs/dns/function.importDnsName.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+```ts
+import { createPublicClient, createWalletClient, http, custom } from 'viem'
+import { mainnet } from 'viem/chains'
+import {
+  addContracts,
+  prepareDnsImport,
+  importDnsName,
+} from '@ensdomains/ensjs'
+
+const [mainnetWithEns] = addContracts([mainnet])
```
[the remark this answers is not in the document]
Comment (TateB) [reply docs/dns/function.importDnsName.md]: this was the api before i tweaked it, fixed

File: packages/ensjs/package.json [hunk 3]
```diff
@@ -1,86 +1,107 @@
...
     "test:watch": "jest --watch --run-in-band",
-    "test:specific": "ts-node --files ",
     "clean": "rm -rf ./dist ./README.md ./LICENSE",
     "lint": "eslint ./src/* --no-error-on-unmatched-pattern",
-    "build:types": "tsc --emitDeclarationOnly --project tsconfig.build.json",
-    "build": "pnpm clean && pnpm lint && pnpm build:types && pnpm node esbuild.js",
+    "build:cjs": "tsc --project tsconfig.build.json --module commonjs --outDir ./dist/cjs --removeComments --verbatimModuleSyntax false && echo > ./dist/cjs/package.json '{\"type\":\"commonjs\"}'",
+    "build:esm": "tsc --project tsconfig.build.json --module es2022 --outDir ./dist/esm && echo > ./dist/esm/package.json '{\"type\":\"module\",\"sideEffects\":false}'",
+    "build:types": "tsc --project tsconfig.build.json --module esnext --declarationDir ./dist/types --emitDeclarationOnly --declaration --declarationMap",
+    "build": "pnpm run clean && pnpm run build:cjs && pnpm run build:esm && pnpm run build:types",
```
[the remark this answers is not in the document]
Comment (TateB) [reply packages/ensjs/package.json:92]: i could make it part of the publish process

File: docs/dns/function.prepareDnsImport.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+---
+
+> [dns](README.md) > prepareDnsImport
+
+# Function: prepareDnsImport()
+
+> **prepareDnsImport**(`client`, `parameters`): `Promise`\< `PrepareDnsImportReturnType` \>
+
+Creates prepared data for `importDnsName()`
```
[the remark this answers is not in the document]
Comment (TateB) [reply docs/dns/function.prepareDnsImport.md]: this is the old function

File: docs/public/function.getAvailable.md (hunk 1 above)
Comment (TateB) [reply docs/public/function.getAvailable.md]: yep the docs shouldnt say that its from the main export, fixed them

File: packages/ensjs/src/functions/public/getRecords.ts [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+  key: string | number
+  call: SimpleTransactionRequest
+  type: 'coin' | 'text' | 'contentHash' | 'abi'
+}
+
+const encode = (
+  client: ClientWithEns,
+  { name, resolver, records }: GetRecordsParameters,
+): TransactionRequestWithPassthrough => {
+  const calls: (CallObj | null)[] = []
```
[the remark this answers is not in the document]
Comment (TateB) [reply packages/ensjs/src/functions/public/getRecords.ts:109]: reduce over what array?

[the remark this answers is not in the document]
Comment (LeonmanRolls) [reply packages/ensjs/src/functions/public/getRecords.ts:109]: keys of `records` in this case

File: packages/ensjs/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (TateB) [reply packages/ensjs/package.json:92]: the errors in generated docs will always come from jsdoc

[the remark this answers is not in the document]
Comment (TateB) [reply packages/ensjs/package.json:92]: unless you mean errors in the actual generation process

File: docs/public/function.getPrice.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+import { mainnet } from 'viem/chains'
+import { addEnsContracts } from '@ensdomains/ensjs'
+import { getPrice } from '@ensdomains/ensjs/public'
+
+const client = createPublicClient({
+  chain: addEnsContracts(mainnet),
+  transport: http(),
+})
+const result = await getPrice(client, { nameOrNames: 'ens.eth' })
+// { base: 352828971668930335n, premium: 0n }
```
[the remark this answers is not in the document]
Comment (TateB) [reply docs/public/function.getPrice.md:69]: do you mean formatted like "0.02 ETH" ?

[the remark this answers is not in the document]
Comment (LeonmanRolls) [reply docs/public/function.getPrice.md:69]: yup

File: packages/ensjs/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (LeonmanRolls) [reply packages/ensjs/package.json:92]: yea okay fair unlikely that the generated docs will have errors

File: docs/public/function.getPrice.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (TateB) [reply docs/public/function.getPrice.md:69]: i think its fine unformatted only tbh, then if they want the value they can use either viem's built-ins or their own formatting func

**Pull request merged**